compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -360,7 +360,16 @@ class Namer { typer: Typer =>
   /** Expand tree and store in `expandedTree` */
   def expand(tree: Tree)(implicit ctx: Context): Unit = tree match {
     case mdef: DefTree =>
-      val expanded = desugar.defTree(mdef)
+      val expanded = mdef match {
+        case mdef: MemberDef if ctx.macrosEnabled && macros.hasAnnotMacros(mdef) =>
+          macros.expandAnnotMacro(mdef) match {
+            case Block(stats, _) =>                 // introducing of same level symbols
```

Comment:
I think it would be better we isolated this point in `expandMacro`. `Namer` should deal with `Thicket`, not with `Block`. It's very important to have only the cleanest representations in core modules such as `Namer` or `Typer`, or we will quickly drown in complexity. So any dirt should to be taken care of at the interfaces if possible.